core/services/subscription_test.go

```diff
@@ -307,6 +315,7 @@ func TestServices_NewInitiatorSubscription_EthLog_ReplayFromBlock(t *testing.T)
 			defer cleanup()
 
 			ethClient := new(mocks.Client)
+			defer ethClient.AssertExpectations(t)
```

Comment:
Hmm, what does this do wrt line numbers when expectations fail? Does it show from the end of the block or from the defer, if it's from the defer I think it could be a little confusing.

Reply:
Its pretty nice actually - if your code makes an extra call it tells you where the extra call was, if it misses a call it tells you where the assertion was e.g. 
```
    job_subscriber_test.go:170: FAIL:	FilterLogs(string,string)
        		at: [job_subscriber_test.go:153]
```